allennlp/modules/token_embedders/embedding.py

```diff
@@ -427,6 +426,10 @@ def __init__(self,
 
             # All the python packages for compressed files share the same interface of io.open
             extension = get_file_extension(main_file_uri)
+            
+            if ".bz2" in extension:
+	            import bz2  # import only when necessary
+            
             package = {
                     '.txt': io,
                     '.vec': io,
```

Comment:
Oops, I approved too soon - this is causing the tests to fail, as `bz2` is undefined here unless the `if` above is triggered.  Can you fix this?

Reply:
Hi @matt-gardner. Now that I have tested on another system which is also an Ubuntu 16.04, I see that the Python 3.6 and above makes it a bit problematic with both the bz2 and lzma libraries. I had to recompile Python several times with no luck. Therefore I suggest dropping the support for these two libraries altogether. 

I believe this would help increase the use of repo. I did remove these two dependencies in my system and made the tests pass. If you agree that these two file types is not crucial for the repo at the moment, I can send another pull request that would drop the dependencies.